### ISSUE TITLE
`auto_generate_links` does not work

### ISSUE THREAD

**Issue by soheilred** [vimwiki/vimwiki#1055]
I am using `vim 8.2` with this vimwiki config in my `.vimrc`
```
let g:vimwiki_list = [{'path': '$HOME/vimwiki/',                                
                     \ 'syntax': 'markdown', 'ext': '.md',                        
                     \ 'auto_diary_index': 1, 'auto_generate_links': 1,          
                     \ }]
```
Based on the README page, where it said
```
auto_diary_index  0                 0, 1

Description~
Set this option to 1 to automatically update the diary index when opened.
```
I expect to see my diary page gets updated every time that I create a new note and save it. But, it doesn't. Any idea why?

**Comment by tinmarino** [comment]
You must create adiary page (special name and path)  with VimwikiMakeDiaryPage.

**Comment by soheilred** [comment]
> You must create adiary page (special name and path) with VimwikiMakeDiaryPage.

Sorry that I didn't clear that up in my original post. I do have a `diary` directory in my `vimwiki` folder and a `diary.md` file in `diary`. Also, the command `VimwikiGenerateLinks` works just file. It does update my `diary.md` file with the recent changes. But, putting `auto_generate_links` in the config doesn't make this process automatic.

**Comment by shadowmax31** [comment]
It's happening on my computer too, my index does not refresh after I create a new entry with VimwikiMakeDiaryPage, but it works when I refresh it manually

I'll try to fix the issue, unless someone is already working on it.

Also, I think that the auto update is linked to the auto_diary_index property and not to auto_generate_links. Or I don't understand the initial issue correctly and we might have two problems...either way I'll check this out

**Comment by shadowmax31** [comment]
Until a real solution is found, as a workaround, you can add this to your .vimrc file

`autocmd BufEnter diary.md :VimwikiDiaryGenerateLinks`

From what I've been testing, the issue seems to be coming from the function `vimwiki#diary#goto_diary_index` in the file `diary.vim`. The function isn't called when you go back to the diary_index. It is not called either when you delete a diary page and are moved back to the index.

Here is the section that should work, but isn't called:

```
  if vimwiki#vars#get_wikilocal('auto_diary_index')
    call vimwiki#diary#generate_diary_section()
    write! " save changes
  endif
```
I'll try to keep looking around, but I don't any more time right now...

**Comment by soheilred** [comment]
>Also, I think that the auto update is linked to the auto_diary_index property and not to auto_generate_links. Or I don't understand the initial issue correctly and we might have two problems...either way I'll check this out
Yes, you're right. auto_diary_index is the issue. I've fixed the title.

**Comment by infinitewhileloop** [comment]
Add this to your `.vimrc`, to automatically update _and_ save the diary index.

```vim
function! s:Fix_auto_diary_index()
  VimwikiDiaryGenerateLinks 
  write
endfunction

augroup vimwiki_fix_auto_diary_index
  autocmd!
  autocmd BufEnter diary.md call <SID>Fix_auto_diary_index()
augroup END
```

**Comment by foozzi** [comment]
Hi there,
Is the issue still not fixed by the author?

**Comment by foozzi** [comment]
I have reviewed the code and noticed that it's not an issue.

Look at this code. It works only manually:

https://github.com/vimwiki/vimwiki/blob/dev/plugin/vimwiki.vim#L358

https://github.com/vimwiki/vimwiki/blob/dev/plugin/vimwiki.vim#L392

https://github.com/vimwiki/vimwiki/blob/dev/plugin/vimwiki.vim#L416

`VimwikiDiaryIndex` runs the code below:

https://github.com/vimwiki/vimwiki/blob/dev/autoload/vimwiki/diary.vim#L336

```vim
function! vimwiki#diary#goto_diary_index(wnum)
  " Jump to diary index of 1. <Int> wikinumber
  " if wnum = 0 the current wiki is used
  if a:wnum == 0
    let idx = vimwiki#vars#get_bufferlocal('wiki_nr')
    if idx < 0  " not in a wiki
      let idx = 0
    endif
  else
    let idx = a:wnum - 1 " convert to 0 based counting
  endif

  if a:wnum > vimwiki#vars#number_of_wikis()
    call vimwiki#u#error('Wiki '.a:wnum.' is not registered in g:vimwiki_list!')
    return
  endif

  call vimwiki#base#edit_file('e', s:diary_index(idx), '')

  if vimwiki#vars#get_wikilocal('auto_diary_index')
    >>> call vimwiki#diary#generate_diary_section()
    write! " save changes
  endif
endfunction
``` 

That's all the mentions of this function.

```bash
grep -r "generate_diary_section" ./
.//plugin/vimwiki.vim:      \ call vimwiki#diary#generate_diary_section()
.//autoload/vimwiki/diary.vim:    call vimwiki#diary#generate_diary_section()
.//autoload/vimwiki/diary.vim:function! vimwiki#diary#generate_diary_section() abort
```

**Comment by fancsali** [comment]
It seems to be work for me on the latest version:

```
Version: 2024.01.24
Os: Linux
Vim: 901
Branch: HEAD
Revision: fde35bb
Date: 2024-01-24 20:14:13 -0700
```
